lib/modules/datasource/endoflife-date/schema.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from 'zod';
+import type { Release } from '../types';
+
+const EndoflifeDateVersionScheme = z
+  .object({
+    cycle: z.string(),
+    releaseDate: z.optional(z.string()),
+    eol: z.optional(z.union([z.string(), z.boolean()])),
+    latest: z.optional(z.string()),
+    link: z.optional(z.string()),
+    lts: z.optional(z.union([z.string(), z.boolean()])),
+    support: z.optional(z.union([z.string(), z.boolean()])),
+    discontinued: z.optional(z.union([z.string(), z.boolean()])),
```

Comment:
why it can be string or boolean?

we should return `deprecated=true` when deprecated or EOL.

Reply:
`discontinued`, `support` and `eol` are all either a date string, e.g. `"2022-12-17"` or a boolean if no exact date information is available.

Reply:
I did not include the `support` field in the determination of the deprecation since some of the support periods end before the product EOL, see e.g. Grafana: https://github.com/endoflife-date/endoflife.date/blob/5874cafbfc42d3e4dee0f62b6da7b11de7cb3657/products/grafana.md?plain=1#L20-L82

Reply:
https://github.com/renovatebot/renovate/blob/9a87c3f7da532bb5ac24d759ae502272aec4fa4e/lib/modules/datasource/types.ts#L47

set `isDeprecated` if
- `support` is true or date has passed by
- same for `eol`

Reply:
https://github.com/renovatebot/renovate/blob/9a87c3f7da532bb5ac24d759ae502272aec4fa4e/lib/modules/datasource/types.ts#L48

set `isStable` to false if some `lts` is true. Do you saw a case when thats a date and if yes it's start or end date?

Reply:
@viceice 

> set isDeprecated if support is true or date has passed by

As mentioned above, I did explicitly not include `support` in the `isDeprecated` decision since the support date sometimes ends before the product EOL (see the linked grafana example).

> set isStable to false if some lts is true. Do you saw a case when thats a date and if yes it's start or end date?

Why should we consider LTS versions as not stable? If it's a date, it's the start date. That's documented in https://endoflife.date/docs/api, although I did not yet check for an example.